### PR TITLE
fix: {data:null} is accepted as valid relationship object; updated packages

### PR DESCRIPTION
**Pull request by FraKraBa**
fix: {data:null} is accepted as valid relationship object; updated packages

### CODE REVIEW

File: test/tests.spec.ts [hunk 1]
```diff
@@ -307,6 +307,12 @@ describe('A JSON:API compound document', () => {
           },
           tests: {
             data: [{ id: '11', type: 'tests' }, { id: '12', type: 'tests' }]
+          },
+          nothing_list: {
```
Comment (danielmenzel) [inline test/tests.spec.ts]:
```suggestion
          nothingList: {
```

File: test/tests.spec.ts [hunk 2]
```diff
@@ -307,6 +307,12 @@ describe('A JSON:API compound document', () => {
...
+            data: []
+          },
+          nothing_detail: {
```
Comment (danielmenzel) [inline test/tests.spec.ts]:
```suggestion
          nothingDetail: {
```

File: test/tests.spec.ts [hunk 3]
```diff
@@ -474,6 +480,15 @@ describe('A JSON:API compound document', () => {
       JsonApi.CardinalityError
     );
   });
+
+  it('empty related data is valid', async () => {
+    const document = await makeDocument(documentPath, testApi);
+    const article = document.resource;
+    const nothing_list = await article.relatedResources['nothing_list'];
```
Comment (danielmenzel) [inline test/tests.spec.ts]:
```suggestion
    const nothingList = await article.relatedResources['nothingList'];
```

File: test/tests.spec.ts [hunk 4]
```diff
@@ -474,6 +480,15 @@ describe('A JSON:API compound document', () => {
...
+    expect(nothing_list).toEqual([]);
+    const nothing_detail = await article.relatedResource['nothing_detail'];
```
Comment (danielmenzel) [inline test/tests.spec.ts]:
```suggestion
    const nothingDetail = await article.relatedResource['nothingDetail'];
```

File: test/tests.spec.ts [hunk 5]
```diff
@@ -474,6 +480,15 @@ describe('A JSON:API compound document', () => {
       JsonApi.CardinalityError
     );
   });
+
+  it('empty related data is valid', async () => {
+    const document = await makeDocument(documentPath, testApi);
+    const article = document.resource;
+    const nothing_list = await article.relatedResources['nothing_list'];
+    expect(nothing_list).toEqual([]);
```
Comment (danielmenzel) [inline test/tests.spec.ts]:
```suggestion
    expect(nothingList).toEqual([]);
```

File: test/tests.spec.ts [hunk 6]
```diff
@@ -474,6 +480,15 @@ describe('A JSON:API compound document', () => {
...
+    const nothing_detail = await article.relatedResource['nothing_detail'];
+    expect(nothing_detail).toEqual(null);
```
Comment (danielmenzel) [inline test/tests.spec.ts]:
```suggestion
    expect(nothingDetail).toEqual(null);
```

File: test/tests.spec.ts [hunk 7]
```diff
@@ -474,6 +480,15 @@ describe('A JSON:API compound document', () => {
       JsonApi.CardinalityError
     );
   });
+
+  it('empty related data is valid', async () => {
```
Comment (danielmenzel) [inline test/tests.spec.ts]:
```suggestion
  it('may contain empty or null resource linkage', async () => {
```

**Comment by FraKraBa** [comment]
removed the "typedoc" package update from this PR

**Pull request merged**